docs/architecture/adr-038-state-listening.md

```diff
@@ -241,6 +241,131 @@ type StreamingService interface {
     ABCIListener
```

Comment:
I find this pretty confusing. You have `ABCIListener` embedded and then you also define all these other `Listen*` ABCI methods with slightly different method signatures...